csrc/host_ir/executor.cpp

```diff
@@ -88,6 +89,45 @@ bool HostIrExecutor::isCompiled() const {
   return (bool)host_ir_container_;
 }
 
+namespace {
+// Validates the sizes and strides of the input and output tensors
+// against the tensorviews and posts the communication.
+c10::intrusive_ptr<c10d::Work> validateAndPostSingleCommunication(
```

Comment:
I'd unwrap this and move validation before the callsites of postSingleCommunication. expr_eval is only needed for validation, and communication, my_device_index, backend are only needed for posting. Wrapping them enforces the caller to feed in a union of two parameter lists, which makes the wrapper a bit cumbersome to use.